app/api/databaseInteraction/findURL.js

```diff
@@ -1,23 +1,27 @@
 var winston = require('winston');
+var MongoClient = require('mongodb').MongoClient;
 
 //	Search the database for the short_url or the original_url
 // depending on the key
-module.exports = function(db, key, paramUrl, callback) {
-	winston.log('info', 'Querying...');
-	// get the sites collection
-	var sites = db.collection('sites');
-	// set key for findOne function
-	var query = {};
-	query[key] = paramUrl;
-	// find document with url the user passed
-	sites.findOne(query, function(e, result) {
-		if (e) {
-			winston.log('error', e);
-		}	else if (result) {
-			callback(result);
-		}	else {
-			winston.log('info', 'Matching query does not exist');
-			callback(null);
-		}
+module.exports = function(dbURL, key, paramUrl) {
+	return new Promise(function(resolve, reject) {
+		MongoClient.connect(dbURL)
+		.then(function(db) {
+			winston.log('info', 'Querying...');
+			// get the sites collection
+			var sites = db.collection('sites');
+			// set key for findOne function
+			var query = {};
+			query[key] = paramUrl;
+
+			sites.findOne(query).then(function(value) {
```

Comment:
good question. In web we do use polyfills to cover a fuller Promise specification. I'm not sure what the convention for this is in serverside node. I've used bluebird in the past and that's totally acceptable.